functions/go/render-helm-chart/third_party/sigs.k8s.io/kustomize/api/builtins/HelmChartInflationGenerator.go

```diff
@@ -291,21 +388,65 @@ func (p *HelmChartInflationGeneratorPlugin) templateCommand() []string {
 	return args
 }
 
-func (p *HelmChartInflationGeneratorPlugin) pullCommand() []string {
+func (p *HelmChartInflationGeneratorPlugin) repoAddArgs() ([]string, string) {
+	if p.Repo != "" {
+		hash := md5.Sum([]byte(p.Repo))
+		strHash := hex.EncodeToString(hash[:])
+		args := []string{
+			"repo",
+			"add",
+			strHash,
+			p.Repo,
+		}
+		if p.password != "" && p.username != "" {
+			args = append(args, []string{
+				"--password", p.password,
+				"--username", p.username,
+			}...)
+		}
+		return args, strHash
+	}
+	return nil, ""
+}
+
+func (p *HelmChartInflationGeneratorPlugin) registryLoginArgs() []string {
+	if p.Repo != "" {
+		args := []string{
```

Comment:
from the latest discussion with CS team. does this function still need the login run?

Reply:
It's not needed for CS because they are using it in exec mode. But it will be needed for the general case for passing credentials to the containerized function.

Reply:
yeah, I agree with still supporting the authentication. I'm wondering whether the `helm registry login` is needed with the password and user flags

Reply:
`helm registry login` is still needed, I don't think `helm pull` with OCI works even with user and password flags unless you do `helm registry login` first.